### PR TITLE
fix(client): modernize stripe form

### PR DESCRIPTION
**Pull request by ahmaxed**
This pr migrates our client stripe implementation from react-stripe-elements to @stripe/react-stripe-js library and uses hooks to manage state.

### CODE REVIEW

**Review by raisedadead — DISMISSED** [review]

**Comment by scissorsneedfoodtoo** [comment]
Tested this locally with the test cards in the Stripe docs and everything LGTM.

**Comment by ahmadabdolsaheb** [comment]
There was an issue on the cert page in which the font color was being rendered based on the user selection. I resolved it with the last push.

**Review by scissorsneedfoodtoo — DISMISSED** [review]
Thanks for catching that issue with the font color on the certification page, @ahmadabdolsaheb. I didn't notice that the last time around.

Went through and made test donations on the donate page and with the popup again. Also confirmed that the certification page looks good with night mode turned on and off. Everything LGTM :+1: :+1:

File: client/src/components/Donation/StripeCardForm.js [hunk 1]
```diff
@@ -1,88 +1,159 @@
...
+        fontSize: '18px',
+        color: `${theme === 'night' ? '#fff' : '#0a0a23'}`
+      }
+    }
+  };
+
+  const handleSubmit = async event => {
+    event.preventDefault();
+
+    if ((!isEmailValid, !isPaymentInfoValid()))
```
Comment (ojeytonwilliams) [inline client/src/components/Donation/StripeCardForm.js:91]:
```suggestion
    if (!isEmailValid || !isPaymentInfoValid())
```
This was quite interesting - I didn't know there was a [comma operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Comma_Operator)

However, I think we just wanted it to be invalid if either of the email or payment info were invalid, right?

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
Hey Ahamad, since Kris has gone through and checked the functionality, I mostly concentrated on stylistic changes.  I found a few things that might help.

File: client/src/components/Donation/StripeCardForm.js (hunk 1 above)
Comment (ahmadabdolsaheb) [reply client/src/components/Donation/StripeCardForm.js:91]: right. thanks for looking into this.

**Review by ojeytonwilliams — APPROVED** [review]
LGTM 👍

**Pull request merged**